### PR TITLE
Rename reflection `Class[?]` to `Type` conversion

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

**Comment by nicolasstucki** [comment]
Based on #9797

File: library/src/scala/tasty/Reflection.scala [hunk 1]
```diff
@@ -1353,8 +1353,13 @@ trait Reflection extends reflect.Types { reflectSelf: CompilerInterface =>
...
       qtype.asInstanceOf[scala.internal.quoted.Type[TypeTree]].typeTree.tpe
 
-    def apply(clazz: Class[_])(using ctx: Context): Type =
-      reflectSelf.Type_apply(clazz)
+    /** Returns the type of the runtime class. This type is the erased representation
+     *  that of the type that is used by arrays.
+     *
+     *  See `Type.of` to get the true type.
+     */
+    def ofErasedClass(clazz: Class[_])(using ctx: Context): Type =
```
Comment (sjrd) [inline library/src/scala/tasty/Reflection.scala:1361]: Is this actually used by anything else than `scala.quoted.ClassLiftable`? If not, could we make it `private[scala]` unless and until we discover an imperative use case outside?

Comment (nicolasstucki) [reply library/src/scala/tasty/Reflection.scala:1361]: Good point.

Comment (nicolasstucki) [reply library/src/scala/tasty/Reflection.scala:1361]: I don't see any uses outside lifting arrays

**Review by sjrd — APPROVED** [review]
I like this much better!

**Pull request merged**